skyportal/plot.py

```diff
@@ -717,22 +716,30 @@ def spectroscopy_plot(obj_id, spec_id=None):
 
     color_map = dict(zip([s.id for s in spectra], viridis(len(spectra))))
 
-    data = pd.concat(
-        [
-            pd.DataFrame(
-                {
-                    'wavelength': s.wavelengths,
-                    'flux': s.fluxes,
-                    'id': s.id,
-                    'telescope': s.instrument.telescope.name,
-                    'instrument': s.instrument.name,
-                    'date_observed': s.observed_at.date().isoformat(),
-                    'pi': s.assignment.run.pi if s.assignment is not None else "",
-                }
-            )
-            for i, s in enumerate(spectra)
-        ]
-    )
+    data = []
+    for i, s in enumerate(spectra):
+
+        # normalize spectra to a common total flux (facilitates easy visual comparison)
+        normfac = np.sum(np.gradient(s.wavelengths) * s.fluxes)
+
+        if not (np.isfinite(normfac) and normfac > 0):
+            # otherwise normalize the value at 6000A to 2e-4
+            minimum_wave_index = np.argmin(np.abs(s.wavelengths - 6000.0))
+            normfac = s.fluxes[minimum_wave_index] / 2.0e-4
```

Comment:
I do think it's good to normalize the spectra so the y-axis isn't scaled to such a small value (since the DC level is arbitrary), but it seems like 2e-4 is a hard-coded factor that is likely related to the wavelength gradient size. A lower or higher-resolution spectrum would likely lead to a different factor. For generality it's worth calculating this factor.

Reply:
Thanks, I totally agree. I did the following in 3f77297: 

1. Instead of normalizing to a common total flux, I normalized to a common average flux per resolving element of 1 across spectra. So,  the values within a given spectrum should average to 1. 

2. In the case where a common average flux per resolving element is negative or undefined, I normalize the value of the spectrum at the median wavelength to be 1.